packages/gantt/backend/sirius-components-collaborative-gantt/src/main/java/org/eclipse/sirius/components/collaborative/gantt/service/GanttTaskService.java

```diff
@@ -165,4 +166,32 @@ private Optional<GanttDescription> findGanttDescription(String ganttDescriptionI
         return this.representationDescriptionSearchService.findById(editingContext, ganttDescriptionId).filter(GanttDescription.class::isInstance).map(GanttDescription.class::cast);
     }
 
+    @Override
+    public IPayload dropTask(DropGanttTaskInput dropTaskInput, IEditingContext editingContext, Gantt gantt) {
+        IPayload payload = new ErrorPayload(dropTaskInput.id(), "Drop task failed");
+
+        Optional<Task> droppedTaskOpt = this.findTask(task -> Objects.equals(task.id(), dropTaskInput.droppedTaskId()), gantt.tasks());
+        Optional<Task> targetTaskOpt = this.findTask(task -> Objects.equals(task.id(), dropTaskInput.targetTaskId()), gantt.tasks());
+        Optional<GanttDescription> ganttDescriptionOpt = this.findGanttDescription(gantt.descriptionId(), editingContext);
+
+        if (droppedTaskOpt.isPresent() && ganttDescriptionOpt.isPresent()) {
+            String targetObjectId = targetTaskOpt.map(task -> task.targetObjectId()).orElseGet(() -> gantt.targetObjectId());
```

Comment:
lambdas could be replaced by methode reference 
`String targetObjectId = targetTaskOpt.map(Task::targetObjectId).orElseGet(gantt::targetObjectId);`